### PR TITLE
`zenith push` prototype

### PR DESCRIPTION
**Pull request by patins**
Haven't gotten around to tests yet, as it involves spawning multiple pageservers. However, I have been able to test manually

### CODE REVIEW

File: pageserver/src/repository.rs [hunk 1]
```diff
@@ -176,6 +178,18 @@ pub trait Timeline {
 
     /// Get vector of prepared twophase transactions
     fn get_twophase(&self, lsn: Lsn) -> Result<Vec<TransactionId>>;
+
+    fn make_sender<'t>(&'t self) -> Box<dyn TimelineSender + 't>;
+}
+
+pub trait TimelineSender<'t> {
```
Comment (hlinnaka) [inline pageserver/src/repository.rs:185]: This interface seems really low level. What is the content of a "message"?

I'm working on replacing the rocksdb implementation at https://github.com/zenithdb/zenith/pull/79, so I'm hoping that we don't introduce any new stuff that is tightly coupled with the current rocksdb implementation :-)

**Review by kelvich — APPROVED** [review]
Thanks a lot! Looks good.

> Haven't gotten around to tests yet, as it involves spawning multiple pageservers.

Indeed. And pageserver start is quite messy right now. I suggest merge this in a current form without test; add config file support to pageserver in separate commit; create infrastructure to perform tests with several pageservers.

**Comment by patins** [comment]
Yep that test plan sounds good to me!

**Comment by kelvich** [comment]
I've put some thoughts on pageserver start/init in https://github.com/zenithdb/zenith/issues/224

**Pull request merged**